photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala

```diff
@@ -114,32 +117,37 @@ class Driver(val params: Params, val sparkContext: SparkContext, val logger: Log
     // Load the model from HDFS, ignoring the feature index loader
     val (gameModel, _) =
       ModelProcessingUtils.loadGameModelFromHDFS(Some(featureShardIdToIndexMapLoader), gameModelInputDir, sparkContext)
+    gameModel.persist(StorageLevel.FREQUENT_REUSE_RDD_STORAGE_LEVEL)
 
-    logger.debug(s"Loaded game model summary:\n${gameModel.toSummaryString}")
+    if (logDatasetAndModelStats) {
+      logger.debug(s"Loaded game model summary:\n${gameModel.toSummaryString}")
+    }
 
-    val scores = gameModel.score(gameDataSet).persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL).materialize()
+    val scores = gameModel.score(gameDataSet)
+      .persistRDD(StorageLevel.FREQUENT_REUSE_RDD_STORAGE_LEVEL).materialize()
 
+    gameDataSet.unpersist()
     gameModel.unpersist
     scores
   }
 
   /**
    * Save the computed scores to HDFS with auxiliary info
    *
-   * @param gameDataSet The GAME data set
    * @param scores The computed scores
    */
-  protected def saveScoresToHDFS(
-      gameDataSet: RDD[(Long, GameDatum)],
-      scores: KeyValueScore): Unit = {
+  protected def saveScoresToHDFS(scores: KeyValueScore): Unit = {
 
     // Take the offset information into account when writing the scores to HDFS
```

Comment:
I would put comments like that in the function comment block, as it is an important piece of information for users: it specifies the contract for the scores on HDFS.